barman/backup.py

```diff
@@ -1628,3 +1634,62 @@ def verify_backup(self, backup_info):
             output.error(e.args[0]["err"])
             return
         output.info(pg_verifybackup.get_output()[0].strip())
+
+    def combine_backups(self, backup_info, backup_name=None, dest_dir=None):
+        """
+        Combines a backup chain into a single synthetic backup using the
+        pg_combinebackup utility.
+
+        :param barman.infofile.LocalBackupInfo backup_info: the last backup
+            in the chain.
+        :param str dest_dir: the directory where the synthetic backup is going
+            to me mounted on. If not passed, the server backup directory is used
+            and the synthetic backup will be a new backup in the backup catalog.
+        """
+        pass
+
+    def create_backup_info(self, backup_info, synthetic_backup_dir):
+        """
+        Generates a backup_info file for a synthetic backup.
+
+        :param barman.infofile.LocalBackupInfo backup_info: the last incremental
+            backup of the chain used to generate the synthetic backup.
+        :param str synthetic_backup_dir: the synthetic backup directory
+        """
+        synthetic_backup_info = SynthethicBackupInfo(
+            server=self.server,
+            backup_dir=synthetic_backup_dir,
+        )
+
+        # set server parameters
+        strategy = PostgresCombineBackupStrategy(self.server.postgres, self.config.name)
+        strategy.set_pg_metadata(synthetic_backup_info)
+
+        # set wal-related parameters from the last incremetal backup
+        wal_fields = (
+            "begin_wal",
+            "begin_xlog",
+            "begin_offset",
+            "end_wal",
```

Comment:
I would say from the test i did that begin_wal, begin_xlog its correct to take from the last incremental! Would this be correct for the others as well?